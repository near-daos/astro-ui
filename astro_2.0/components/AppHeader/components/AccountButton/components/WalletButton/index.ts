export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletButton/WalletButton';
