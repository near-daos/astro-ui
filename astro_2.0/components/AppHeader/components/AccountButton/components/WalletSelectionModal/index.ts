export * from 'astro_2.0/components/AppHeader/components/AccountButton/components/WalletSelectionModal/WalletSelectionModal';
