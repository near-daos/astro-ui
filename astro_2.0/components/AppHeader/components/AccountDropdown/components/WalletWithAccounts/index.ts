export * from 'astro_2.0/components/AppHeader/components/AccountDropdown/components/WalletWithAccounts/WalletWithAccounts';
