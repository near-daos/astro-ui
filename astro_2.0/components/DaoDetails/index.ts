export * from './DaoDetailsMinimized';
export * from './DaoDetailsGrid';
