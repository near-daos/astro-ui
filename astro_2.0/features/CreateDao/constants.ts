export const STEPS = {
  INFO: 'info',
  KYC: 'kyc',
  LINKS: 'links',
  MEMBERS: 'members',
  PROPOSALS: 'proposals',
  ASSETS: 'assets',
};
