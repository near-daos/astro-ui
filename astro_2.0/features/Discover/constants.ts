import { Interval } from 'services/DaoStatsService/types';

export const CONTRACT = { contract: 'astro' };

export enum DaoStatsTopics {
  GENERAL_INFO = 'generalInfo',
  USERS_AND_ACTIVITY = 'usersAndActivity',
  GOVERNANCE = 'governance',
  FLOW = 'flow',
  TVL = 'tvl',
  TOKENS = 'tokens',
}

export enum GeneralInfoTabs {
  ACTIVITY = 'activity',
  ACTIVE_DAOS = 'activeDaos',
}

export enum UsersAndActivityTabs {
  ACTIVE_USERS = 'activeUsers',
  ALL_USERS_ON_PLATFORM = 'allUsersOnPlatform',
  USERS_MEMBERS_OF_DAO = 'usersMembersOfDao',
  AVERAGE_NUMBER_OF_USERS_PER_DAO = 'averageNumberOfUsersPerDao',
  NUMBER_OF_INTERACTIONS = 'numberOfInteractions',
  AVERAGE_NUMBER_OF_INTERACTIONS_PER_DAO = 'averageNumberOfInteractionsPerDao',
  ALL_USERS_PER_DAO = 'allUsersPerDao',
}

export enum GovernanceTabs {
  NUMBER_OF_PROPOSALS = 'numberOfProposals',
  VOTE_THROUGH_RATE = 'voteThroughRate',
  ACTIVE_PROPOSALS = 'activeProposals',
  ACTIVE_VOTES = 'activeVotes',
}

export enum FlowTabs {
  TOTAL_IN = 'totalIn',
  TOTAL_OUT = 'totalOut',
  INCOMING_TRANSACTIONS = 'incomingTransactions',
  OUTGOING_TRANSACTIONS = 'outgoingTransactions',
}

export enum TvlTabs {
  PLATFORM_TVL = 'platformTvl',
  VL_IN_BOUNTIES = 'vlInBounties',
  NUMBER_OF_BOUNTIES = 'numberOfBounties',
  VL_OF_BOUNTIES = 'vlOfBounties',
  TVL = 'tvl',
}

export enum TokensTabs {
  NUMBER_OF_FTS = 'numberOfFts',
  VL_OF_FTS = 'vlOfFts',
  NUMBER_OF_NFTS = 'numberOfNfts',
}

export const intervalOptions = [
  { label: 'daily', value: Interval.DAY },
  { label: 'weekly', value: Interval.WEEK },
  { label: 'monthly', value: Interval.MONTH },
];
