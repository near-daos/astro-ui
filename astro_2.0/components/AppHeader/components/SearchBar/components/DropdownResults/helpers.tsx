import styles from 'astro_2.0/components/AppHeader/components/SearchBar/components/DropdownResults/DropdownResults.module.scss';
import { Badge } from 'components/Badge';
import React, { ReactNode } from 'react';
import { Proposal, ProposalType } from 'types/proposal';
import { Button } from 'components/button/Button';
import TextTruncate from 'react-text-truncate';

export function getProposalSearchSummary(
  proposal: Proposal,
  onClick: () => void
): ReactNode {
  let content;

  switch (proposal.kind.type) {
    case ProposalType.AddMemberToRole: {
      content = (
        <div className={styles.summary}>
          <span>
            Add <strong>{proposal.kind.memberId}</strong> as a member to
          </span>
          <Badge size="small">{proposal.kind.role}</Badge>
        </div>
      );
      break;
    }
    case ProposalType.RemoveMemberFromRole: {
      content = (
        <div className={styles.summary}>
          <span>
            Remove <strong>{proposal.kind.memberId}</strong> from
          </span>
          <Badge size="small">{proposal.kind.role}</Badge>
        </div>
      );
      break;
    }
    case ProposalType.ChangePolicy: {
      content = (
        <div className={styles.summary}>
          <span>
            This is a proposal to change voting policies for our groups.
          </span>
        </div>
      );
      break;
    }
    case ProposalType.FunctionCall: {
      content = (
        <div className={styles.summary}>
          <span>Function call</span>
        </div>
      );
      break;
    }
    case ProposalType.Transfer: {
      content = (
        <div className={styles.summary}>
          <span>I would like to request a payment</span>
        </div>
      );
      break;
    }
    case ProposalType.Vote:
    case ProposalType.AddBounty:
    case ProposalType.BountyDone:
    case ProposalType.UpgradeSelf:
    case ProposalType.UpgradeRemote:
    case ProposalType.SetStakingContract:
    default: {
      content = (
        <div className={styles.summary}>
          <TextTruncate
            line={3}
            element="span"
            truncateText="â€¦"
            text={proposal.description}
            textTruncateChild={null}
          />
        </div>
      );
    }
  }

  return (
    <Button
      variant="tertiary"
      onClick={onClick}
      className={styles.row}
      key={proposal.id}
    >
      {content}
    </Button>
  );
}
