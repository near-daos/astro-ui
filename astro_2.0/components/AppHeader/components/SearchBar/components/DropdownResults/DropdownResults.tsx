import isEmpty from 'lodash/isEmpty';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';
import React, { VFC, useCallback } from 'react';
import { useWalletContext } from 'context/WalletContext';

import { SEARCH_PAGE_URL } from 'constants/routing';

import { useSearchResults } from 'features/search/search-results';
import { NoSearchResultsView } from 'features/search/search-results/components/NoSearchResultsView';

import {
  ResultSection,
  SearchResultDaoCard,
  SearchResultPeopleCard,
  SearchResultProposalLine,
} from './components/ResultSection';

import styles from './DropdownResults.module.scss';

interface DropdownResultsProps {
  width: number;
  closeSearch: () => void;
  query: string;
}

export const DropdownResults: VFC<DropdownResultsProps> = ({
  width,
  closeSearch,
  query,
}) => {
  const DAOS_TAB_IDNEX = 0;
  const PROPOSAL_TAB_INDEX = 1;
  const PEOPLE_TAB_INDEX = 2;

  const { accountId } = useWalletContext();
  const { t } = useTranslation('common');

  const router = useRouter();
  const { searchResults } = useSearchResults();

  const { daos, proposals, members } = searchResults || {};

  const goToSearchPage = useCallback(
    (tabIndex: number) => {
      router.push(
        {
          pathname: SEARCH_PAGE_URL,
          query: {
            ...router.query,
            tab: tabIndex,
          },
        },
        undefined,
        {
          shallow: true,
        }
      );
    },
    [router]
  );

  const goToDaosTabOnSearchPage = useCallback(() => {
    goToSearchPage(DAOS_TAB_IDNEX);
  }, [goToSearchPage]);

  const goToProposalsTabOnSearchPage = useCallback(() => {
    goToSearchPage(PROPOSAL_TAB_INDEX);
  }, [goToSearchPage]);

  const goToPeopleTabOnSearchPage = useCallback(() => {
    goToSearchPage(PEOPLE_TAB_INDEX);
  }, [goToSearchPage]);

  function renderNoResults() {
    if (
      isEmpty(daos) &&
      isEmpty(proposals) &&
      isEmpty(members) &&
      query !== ''
    ) {
      return <NoSearchResultsView query={query} />;
    }

    return null;
  }

  function getThreeFirstResults<T>(data?: T[]): T[] {
    return data?.slice(0, 3) || [];
  }

  function renderDaos() {
    const sorted =
      daos?.sort((a, b) => {
        if (a.accountIds.includes(accountId)) {
          return -1;
        }

        if (b.accountIds.includes(accountId)) {
          return 1;
        }

        return 0;
      }) ?? [];

    const dataToRender = getThreeFirstResults(sorted);

    return (
      <ResultSection
        data={sorted}
        title={t('header.search.daos')}
        onSeeAll={goToDaosTabOnSearchPage}
      >
        {dataToRender.map(dao => (
          <SearchResultDaoCard data={dao} key={dao.id} onClick={closeSearch} />
        ))}
      </ResultSection>
    );
  }

  function renderProposals() {
    const proposalsToRender = getThreeFirstResults(proposals);

    return (
      <ResultSection
        title={t('header.search.proposals')}
        data={proposals}
        onSeeAll={goToProposalsTabOnSearchPage}
        contentClassName={styles.proposalsContent}
      >
        {proposalsToRender.map(proposal => (
          <SearchResultProposalLine
            data={proposal}
            key={proposal.id}
            onClick={goToProposalsTabOnSearchPage}
          />
        ))}
      </ResultSection>
    );
  }

  function renderPeople() {
    const peopleToRender = getThreeFirstResults(members);

    return (
      <ResultSection
        title={t('header.search.people')}
        data={members}
        onSeeAll={goToPeopleTabOnSearchPage}
      >
        {peopleToRender.map(person => (
          <SearchResultPeopleCard
            data={person}
            key={person.id}
            onClick={goToPeopleTabOnSearchPage}
          />
        ))}
      </ResultSection>
    );
  }

  return (
    <div className={styles.root} style={{ width }}>
      {renderDaos()}
      {renderProposals()}
      {renderPeople()}

      {renderNoResults()}
    </div>
  );
};
