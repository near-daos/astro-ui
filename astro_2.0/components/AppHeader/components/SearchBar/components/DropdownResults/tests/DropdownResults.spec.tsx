/* eslint-disable @typescript-eslint/ban-ts-comment, @typescript-eslint/no-empty-function */
import get from 'lodash/get';
import { useRouter } from 'next/router';
import { fireEvent } from '@testing-library/dom';

import { render } from 'jest/testUtils';
import { SEARCH_PAGE_URL } from 'constants/routing';
import { useSearchResults } from 'features/search/search-results';

import { DropdownResults } from 'astro_2.0/components/AppHeader/components/SearchBar/components/DropdownResults';

jest.mock('features/search/search-results', () => {
  return {
    useSearchResults: jest.fn(),
  };
});

jest.mock('next/router', () => {
  return {
    useRouter: jest.fn(),
  };
});

describe('search results dropdown', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('Should render component', () => {
    // @ts-ignore
    useSearchResults.mockImplementation(() => ({}));

    const { container } = render(
      <DropdownResults width={100} closeSearch={() => {}} query="asd" />
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render search results', () => {
    const daoName = 'dao 1';
    const proposalDescription = 'proposal description';
    const userName = 'Alex';

    // @ts-ignore
    useSearchResults.mockImplementation(() => ({
      searchResults: {
        daos: [{ displayName: daoName }],
        proposals: [{ description: proposalDescription }],
        members: [{ name: userName }],
      },
    }));

    const component = render(
      <DropdownResults width={100} closeSearch={() => {}} query="asd" />
    );

    expect(component.getAllByText(daoName)).toHaveLength(1);
    expect(component.getAllByText(proposalDescription)).toHaveLength(1);
    expect(component.getAllByText(userName)).toHaveLength(1);
  });

  it.each`
    mockData                                           | tab
    ${{ daos: [{ displayName: 'dao 1' }] }}            | ${0}
    ${{ proposals: [{ description: 'description' }] }} | ${1}
    ${{ members: [{ name: 'Alex' }] }}                 | ${2}
  `('Should go to search page, dao tab', ({ mockData, tab }) => {
    const router = {
      push: jest.fn(),
    };

    // @ts-ignore
    useRouter.mockImplementation(() => router);

    // @ts-ignore
    useSearchResults.mockImplementation(() => ({
      searchResults: mockData,
    }));

    const component = render(
      <DropdownResults width={100} closeSearch={() => {}} query="asd" />
    );

    const seeAllBtn = component.getByText('header.search.seeAll', {
      exact: false,
    });

    fireEvent.click(seeAllBtn);

    expect(router.push).toBeCalled();
    expect(get(router.push, 'mock.calls.0.0')).toStrictEqual({
      pathname: SEARCH_PAGE_URL,
      query: { tab },
    });
  });
});

/* eslint-enable @typescript-eslint/ban-ts-comment, @typescript-eslint/no-empty-function */
