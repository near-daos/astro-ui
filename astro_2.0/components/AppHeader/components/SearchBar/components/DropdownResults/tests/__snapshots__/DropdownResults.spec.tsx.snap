// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search results dropdown Should render component 1`] = `
<div>
  <div
    class="root"
    style="width: 100px;"
  >
    <div
      class="root root"
    >
      <div
        class="image image"
      />
      <div
        class="title"
      >
        No results
      </div>
      <div
        class="desc"
      >
        We couldn't find anything matching your search. Try again with a different term.
      </div>
    </div>
  </div>
</div>
`;
